.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,19 @@
+# To boot the app run the following:
+# docker-compose run auto-gpt
+version: '3.9'
+
+services:
+  auto-gpt:
+    depends_on:
+      - redis
+    build:
+      dockerfile: ./.devcontainer/Dockerfile
+      context: ../
+    tty: true
```

Comment:
what does this do?

Reply:
`tty: true` sets a flag that lets the container know that you're intending to act with 

The `context: ../` sets the context of the build to be one level up from the `.devcontainer` folder, AKA the project root. 

The `dockerfile: ./.devcontainer/Dockerfile` defines the target dockerfile relative to the context of the build. 

The dockerfile in the  `.devcontainer' folder is required for properly setting up the dev enviroment, as it's entry point is a TTY and the user is the default vscode devcontainer user instead of the user defined in the docerkfile of the project root.

Reply:
Haha thanks, the comment was actually only on the `tty: true`.

Reply:
NP, It's just a way to avoid doing the common thing where you have an entry point that is just a keep alive process